2_run/src/run_glm3.R

```diff
@@ -0,0 +1,150 @@
+#'  @param begin the begin date for the model time_period
+#'  @param end the end date for the model_time_period
+#'  @param burn_in length of burn-in period, in days
+#'  @param burn_out length of burn-out period, in days
+#'  @return a munged dataframe of meteorological data that includes
+#'  burn-in and burn-out time
+munge_meteo_w_burn_in_out <- function(meteo_xwalk, begin, end, burn_in, burn_out){
+  meteo_data <- arrow::read_feather(meteo_xwalk$meteo_fl) %>%
+    mutate(time = as.Date(time)) %>% filter(time >= as.Date(begin) & time <= as.Date(end)) %>%
+    mutate(Rain = case_when(Snow > 0 ~ 0, TRUE ~ Rain))
+  # create a burn-in. We're going to mirror the data of the first year:
+  burn_in_data <- meteo_data[2:burn_in, ] %>% arrange(desc(time)) %>%
+    mutate(time = seq(from = meteo_data[1,]$time-burn_in+1, by = 'day', length.out = length(2:burn_in)))
+  # bind with real data, prior to the begin date. We're not duplicating dates, hence the "2"
+  meteo_data <- bind_rows(burn_in_data, meteo_data)
+  # create a burn-out. Mirror the final data:
+  burn_out_data <- tail(meteo_data, burn_out) %>% head(-1L) %>% arrange(desc(time)) %>%
+    mutate(time = seq(from = tail(meteo_data,1)$time+1, by = 'day', length.out = length(2:burn_out)))
+  # bind with real data, at the end of the sequence
+  meteo_data <- bind_rows(meteo_data, burn_out_data)
+  return(meteo_data)
+}
+
+#' @title function to extract and write glm output following a model run
+#' @description Extract temperature and ice data from the model
+#' run and save to the export feather file. 
+#' From Jordan's GLM projection code: 
+#' https://github.com/jread-usgs/lake-temperature-process-models/blob/master/3_run/src/run_glm_utils.R#L89-L101
+#' @param sim_lake_dir the directory for the model simulation
+#' @param nml_obj the complete nml object for the model run
+#' @param export_fl the filepath for the exported feather file
+extract_glm_output <- function(sim_lake_dir, nml_obj, export_fl) {
+  out_dir <- glmtools::get_nml_value(nml_obj, arg_name = 'out_dir')
+  out_fn <- paste0(glmtools::get_nml_value(nml_obj, 'out_fn'), '.nc')
+  nc_filepath = file.path(sim_lake_dir, out_dir, out_fn)
+  lake_depth <- glmtools::get_nml_value(nml_obj, arg_name = 'lake_depth')
+  export_depths <- seq(0, lake_depth, by = 0.5)
+  temp_data <- glmtools::get_temp(nc_filepath, reference = 'surface', z_out = export_depths) %>%
+    mutate(date = as.Date(lubridate::floor_date(DateTime, 'days'))) %>% select(-DateTime)
+  glmtools::get_var(nc_filepath, var_name = 'hice') %>%
+    dplyr::mutate(ice = hice > 0, date = as.Date(lubridate::ceiling_date(DateTime, ' days'))) %>% dplyr::select(-hice, -DateTime) %>%
+    dplyr::left_join(temp_data, ., by = 'date') %>%
+    select(time = date, everything()) %>%
+    feather::write_feather(export_fl)
+}
+
+#' @title run glm3 simulation
+#' @description For each lake - gcm - time period combo, write the nml
+#'  file, the meteo data, and run the glm3 model
```

Comment:
Maybe include the part about this extracting and saving the results using `extract_glm_output()` and cleaning up after itself.